File: experiments/mvp1_validation/fixtures/python-pkg/app/pagination.py
# Fixture file for the HORO-1127 release-gate corpus. Exists only so
# bounded reconnaissance has a real "pagination"-named path to match
# against prompts like "fix the off-by-one in the pagination logic".


def page_bounds(page: int, page_size: int, total: int) -> tuple[int, int]:
    start = page * page_size
    end = min(start + page_size, total)
    return start, end

File: experiments/mvp1_validation/fixtures/rust-crate/src/pagination.rs
// Fixture file for the HORO-1127 release-gate corpus. Exists only so
// bounded reconnaissance has a real "pagination"-named path to match
// against prompts like "fix the off-by-one in the pagination logic".

pub fn page_bounds(page: usize, page_size: usize, total: usize) -> (usize, usize) {
    let start = page * page_size;
    let end = (start + page_size).min(total);
    (start, end)
}

pub fn lib_root() {}
